x-pack/plugins/stack_alerts/server/alert_types/search_threshold/alert_type.ts

```diff
@@ -0,0 +1,246 @@
+     */
+    actionVariables: {
+      context: [
+        { name: 'message', description: actionVariableContextMessageLabel },
+        { name: 'title', description: actionVariableContextTitleLabel },
+        { name: 'date', description: actionVariableContextDateLabel },
+        { name: 'value', description: actionVariableContextValueLabel },
+        { name: 'conditions', description: actionVariableContextConditionsLabel },
+      ],
+      params: [
+        { name: 'threshold', description: actionVariableContextThresholdLabel },
+        { name: 'thresholdComparator', description: actionVariableContextThresholdComparatorLabel },
+      ],
+    },
+    minimumLicenseRequired: 'basic',
+    /**
+     * Whether the rule type is exportable from the Saved Objects Management UI.
+     */
+    isExportable: true,
+    /**
+     * This is a function to be called when executing a rule on an interval basis.
+     */
+    executor,
+    /**
+     * The id of the application producing this rule type.
+     * Since were creating a new stack rule type, we don't set Discover here
+     */
+    producer: STACK_ALERTS_FEATURE_ID,
+    /**
+     * The length of time a rule can run before being cancelled due to timeout
+     */
+    ruleTaskTimeout: '5m',
+    /**
+     * Used for extract and inject saved object references of search source
+     */
+    useSavedObjectReferences: {
+      extractReferences: (params) => {
+        const [searchSourceFields, references] = extractReferences(params.searchSource);
+        const newParams = {
+          ...params,
+          searchSource: searchSourceFields,
+        } as SearchThresholdExtractedParams;
+        return { params: newParams, references };
+      },
+      injectReferences: (params, references) => {
+        return {
+          ...params,
+          searchSource: injectReferences(params.searchSource, references),
+        } as SearchThresholdParams;
+      },
+    },
+  };
+
+  async function executor(
+    options: AlertExecutorOptions<
+      SearchThresholdParams,
+      { previousTimestamp?: string; previousTimeRange?: { from: string; to: string } },
+      {},
+      ActionContext,
+      typeof ActionGroupId
+    >
+  ) {
+    const { name, params, alertId, state } = options;
+    const timestamp = new Date().toISOString();
+    logger.debug(`searchThreshold (${alertId}) previousTimestamp: ${state.previousTimestamp}`);
+    logger.debug(
+      `searchThreshold (${alertId}) searchSource: ${JSON.stringify(params.searchSourceFields)}`
+    );
+
+    const compareFn = ComparatorFns.get(params.thresholdComparator);
+    if (compareFn == null) {
+      throw new Error(
+        i18n.translate('xpack.stackAlerts.searchThreshold.invalidComparatorErrorMessage', {
+          defaultMessage: 'invalid thresholdComparator specified: {comparator}',
+          values: {
+            comparator: params.thresholdComparator,
+          },
+        })
+      );
+    }
+
+    const nrOfDocs = 10;
```

Comment:
The current alert is always active when the when a lower threshold is configured, well, it's a start